sdk/python/packages/flet-core/src/flet_core/template_route.py

```diff
@@ -18,7 +53,15 @@ def match(self, route_template: str) -> bool:
 
         if match:
             self.__last_params = match.groupdict()
-            for k, v in self.__last_params.items():
-                setattr(self, k, v)
+            # optional pattern such as `:id?` will be matched and set as None
+            # if there are defaults provided, we skip overwriting the properties with None
+            skip_none = True if self.__defaults else False
+            self.__set_props(self.__last_params, skip_none=skip_none)
```

Comment:
Second thought. This is, strictly speaking not an incremental change because the behavior of the following code changes:

```python
troute = TemplateRoute(page.route, defaults={"id": get_random_id()})
if troute.match("/books/:name?"):
    print("Book view ID:", troute.name)
else:
    print("Unknown route")
```

In the main branch `troute.name is None` and in this branch `troute.name` does not exist.

I'd like to know if you guys are in favor of this "defaults" idea (whether it's added to the `__init__` or to the `match`), and then I'll update accordingly.

Reply:
You mean your addition breaks some existing scenarios?

Also, why would you need to set book ID to a random number? 😏 Could you provide a real-world scenario?

Reply:
> Could you provide a real-world scenario?

Good question! I admit that I don't have any based on my limited app dev experiences.

I think I'll just close this PR.